Monika After Story/game/zz_consumables.rpy

```diff
@@ -0,0 +1,1400 @@
+
+            #If we have no consumables, then there's no point in doing anything
+            if not available_cons:
+                return
+
+            #If we're currently prepping/having anything, we don't need to do anything else
+            if persistent._mas_current_consumable[_type]["id"] is not None:
+                return
+
+            #Otherwise, step two: what are we having?
+            cons = random.choice(available_cons)
+
+            #Setup some vars
+            _now = datetime.datetime.now()
+
+            #Time to C O N S U M E
+            #First, clear vars so we start fresh
+            MASConsumable._reset(_type)
+
+            #First, should we even have this?
+            if cons.shouldHave():
+                #If we prepare, we prep using 3-5 chages worth (to acct for multiple servings)
+                if cons.prepable():
+                    cons.use(amount=random.randint(3,5))
+
+                #Otherwise, if it's a non-prepable, just one
+                else:
+                    cons.use()
+
+                #Are we loading in after the time? If so, we should already have the cons out. No prep, just have
+                #Though we'll not guarantee this to add a degree of realism/variance (80% chance she'll start with it out)
+                if startup and cons.isLateEntry() and random.randint(1, 100) <= 80:
+                    cons.have(skip_leadin=True)
+
+                else:
+                    #If this is a prepable, we should prep it
+                    if cons.prepable() and cons.shouldPrep(_now):
+                        cons.prepare()
+
+                    #Otherwise, we'll just set up having it
+                    elif not cons.prepable():
+                        cons.have()
+
+#END: MASConsumable class
+
+    #START: Global functions
+    def mas_generateShoppingList(low_cons_list=None):
+        """
+        Generates a list of consumables we're low on in the form of a 'shopping list'
+        and exports it to the characters folder
+
+        IN:
+            low_cons_list - List of MASConsumable objects that we're low on
+            If None, we get it here
+            (Default: None)
+        """
+        #First, get all the consumables we're low on if not provided
+        if low_cons_list is None:
+            low_cons_list = MASConsumable._getLowCons()
+
+        START_TEXT = (
+            "Hi, [player],\n"
+            "Just letting you know I'm running low on a couple of things.\n"
+            "You wouldn't mind getting some more for me, would you?\n\n"
+            "Here's a list of what I'm running out of:\n"
+        )
+
+        MID_TEXT = ""
+
+        END_TEXT = (
+            "Thanks, [player]~"
+        )
+
+        for cons in low_cons:
+            MID_TEXT += "- {0}\n".format(cons.disp_name.capitalize())
+
+        MID_TEXT += "\n"
+
+        shopping_list = open(renpy.config.basedir + "/characters/shopping_list.txt", "w")
```

Comment:
use `with` notation for simple file handling as it already handles close and exceptions:
```python
with open(...) as shopping_list:
    shopping_list.write(...)
```